lib/Doctrine/ORM/Utility/EntityList.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+namespace Doctrine\ORM\Utility;
+
+use Doctrine\Common\Persistence\Mapping\ClassMetadataFactory;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Doctrine\ORM\ORMInvalidArgumentException;
+
+/**
+ * EntityList contains functions that operation on a list of entities
+ *
+ * @since   2.5
+ * @author  Oleg Namaka <avantprogger@gmail.com>
+ */
+class EntityList
+{
+    /**
+     * Sorts an array of entities by order criteria. One level sorting is supported only
+     *
+     * Example usages:
+     *
+     * sort($entityList, array('name' => 'ASC', 'Email' => 'DESC');
+     *
+     * @param array                $entityList Entities to sort
+     * @param array                $orderBy    Order by criteria
+     * @param ClassMetadataFactory $factory    Class metadata factory
+     *
+     * @throws \LogicException
+     * @throws \InvalidArgumentException
+     */
+    public static function sort(array &$entityList, array $orderBy, ClassMetadataFactory $factory)
```

Comment:
IMO, this entire piece of logic should be completely avoided. If we find by identifier, then sorting is not needed, as we already do direct hits from memory